src/arrayMethodSort.js

```diff
@@ -1,11 +1,36 @@
 'use strict';
 
-/**
- * Implement method Sort
- */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  function getComparisonFunction(compareFunction) {
+    if (typeof compareFunction === 'function') {
+      return compareFunction;
+    } else {
+      return function (a, b) {
+        if (a.toString() > b.toString()) {
+          return 1;
+        }
+
+        if (b.toString() > a.toString()) {
+          return -1;
+        }
+
+        return 0;
+      };
+    }
+  }
+
+  [].__proto__.sort2 = function (compareFunction) {
+    const compare = getComparisonFunction(compareFunction);
+
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = i + 1; j < this.length; j++) {
+        if (compare(this[i], this[j]) > 0) {
+          [this[i], this[j]] = [this[j], this[i]];
+        }
+      }
+    }
```

Comment:
You've implemented a bubble sort algorithm, which is correct and meets the task requirements. However, bubble sort is not very efficient for large datasets as it has a time complexity of O(n^2). If you're looking to optimize performance for larger arrays, consider implementing a more efficient sorting algorithm like quicksort or mergesort.